### PR TITLE
Added web.config for Go webapp

### PR DESCRIPTION
**Pull request by shrimpy**

### CODE REVIEW

File: lib/templates/deploy.batch.go.template [hunk 1]
```diff
@@ -65,4 +65,12 @@ ECHO Copy static files for deployment
 call :ExecuteCmd "%KUDU_SYNC_CMD%" -v 50 -f "%DEPLOYMENT_SOURCE%" -t "%DEPLOYMENT_TARGET%" -n "%NEXT_MANIFEST_PATH%" -p "%PREVIOUS_MANIFEST_PATH%" -i ".git;.hg;.deployment;deploy.cmd;*.go"
 IF !ERRORLEVEL! NEQ 0 goto error
 
+ECHO Copy web.config
+IF NOT EXIST "%DEPLOYMENT_TARGET%\web.config" (
+    CP %GO_WEB_CONFIG_TEMPLATE% "%DEPLOYMENT_TARGET%\web.config"
```
Comment (amitapl) [inline lib/templates/deploy.batch.go.template]: cp --> copy.

Comment (amitapl) [inline lib/templates/deploy.batch.go.template]: also %GO_WEB_CONFIG_TEMPLATE% may not exist (when running locally)

Comment (shrimpy) [inline lib/templates/deploy.batch.go.template]: thanks! updated to "copy"

i tried running kudu.web locally and create a new website. i am able to see %GO_WEB_CONFIG_TEMPLATE% from kudu console

and deployment is working fine too see below log "remote: Copy web.config"

is there other scenario i need to be awared?

````batch
E:\kudu\apps\go5>ECHO %GO_WEB_CONFIG_TEMPLATE%
E:\kudu\Kudu.Services.Web\bin\scripts\go.web.config.template
````

````
Counting objects: 3, done.
Delta compression using up to 12 threads.
Compressing objects: 100% (2/2), done.
Writing objects: 100% (2/2), 219 bytes | 0 bytes/s, done.
Total 2 (delta 1), reused 0 (delta 0)
remote: Updating branch 'master'.
remote: Updating submodules.
remote: Preparing deployment for commit id 'bb32cd67d7'.
remote: Generating deployment script.
remote: Running deployment command...
remote: Handling Go deployment.
remote: Prepare workspace
remote: GOROOT F:\Go
remote: Creating GOPATH\bin C:\Users\go3\AppData\Local\Temp\e716f9c6-6966-45f9-8aff-14244ff12312\gopath\bin
remote: Creating GOPATH\pkg C:\Users\go3\AppData\Local\Temp\e716f9c6-6966-45f9-8aff-14244ff12312\gopath\pkg
remote: Creating GOPATH\src C:\Users\go3\AppData\Local\Temp\e716f9c6-6966-45f9-8aff-14244ff12312\gopath\src
remote: Creating C:\Users\go3\AppData\Local\Temp\e716f9c6-6966-45f9-8aff-14244ff12312\gopath\src\azureapp
remote: Copy source code to Go workspace
remote:
remote: -------------------------------------------------------------------------------
remote:    ROBOCOPY     ::     Robust File Copy for Windows
remote: -------------------------------------------------------------------------------
remote:
remote:   Started : Wednesday, April 22, 2015 5:37:53 PM
remote:    Source : E:\kudu\apps\go3\site\repository\
remote:      Dest : C:\Users\go3\AppData\Local\Temp\e716f9c6-6966-45f9-8aff-14244ff12312\gopath\src\azureapp\
remote:
remote:     Files : *.*
remote:
remote: Exc Files : .deployment
remote:             deploy.cmd
remote:
remote:  Exc Dirs : .git
remote:             .hg
remote:
remote:   Options : *.* /NDL /NFL /S /E /DCOPY:DA /COPY:DAT /NP /R:1000000 /W:30
remote:
remote: ------------------------------------------------------------------------------
remote:
remote:
remote: ------------------------------------------------------------------------------
remote:
remote:                Total    Copied   Skipped  Mismatch    FAILED    Extras
remote:     Dirs :         3         1         1         0         0         0
remote:    Files :        10         8         2         0         0         0
remote:    Bytes :     8.5 k     8.5 k         4         0         0         0
remote:    Times :   0:00:00   0:00:00                       0:00:00   0:00:00
remote:
remote:
remote:    Speed :              549250 Bytes/sec.
remote:    Speed :              31.428 MegaBytes/min.
remote:    Ended : Wednesday, April 22, 2015 5:37:53 PM
remote:
remote: Resolving dependencies
remote: ......
remote: Building Go app to produce exe file
remote: Copy static files for deployment
remote: KuduSync.NET from: 'E:\kudu\apps\go3\site\repository' to: 'E:\kudu\apps\go3\site\wwwroot'
remote: Copying file: 'deploy.batch.go.template.cmd'
remote: Copying file: 'index.html'
remote: Copying file: 'index1.txt'
remote: Copying file: 'index2.txt'
remote: Copying file: 'test.cmd'
remote: Copying file: 'bla\index.txt'
remote: Copy web.config
remote: Finished successfully.
remote: Deployment successful.
To http://localhost:7952/go3.git
   4abab10..bb32cd6  master -> master
````

**Comment by amitapl** [comment]
when I say run locally I mean since you can generate the script locally (for custom deployment script scenario) you can also run the script locally (without kudu) from the root of your repository.

**Comment by shrimpy** [comment]
oh i see!

but how about %DEPLOYMENT_TEMP% and other environment variable, would`t they have the same issue?

**Comment by amitapl** [comment]
https://github.com/projectkudu/KuduScript/blob/master/lib/templates/deploy.batch.aspnet.template#L1-L4

**Comment by amitapl** [comment]
maybe we should pull this up to the generic prefix template

**Comment by shrimpy** [comment]
check %GO_WEB_CONFIG_TEMPLATE% in deploy.batch.prefix.template might be overkill, since it is Go specific file. if somehow it is not there, we do not want to cause none-Go deployment failure.

I think i will do another if check in my script, if %GO_WEB_CONFIG_TEMPLATE%  is not defined, i will "goto error"

what do you think?

**Comment by shrimpy** [comment]
oh.. i am already doing it. 

[IF !ERRORLEVEL! NEQ 0 goto error](https://github.com/projectkudu/KuduScript/pull/25/files#diff-d9af631065e6962f13dd16d26801eb6dR71)

**Comment by amitapl** [comment]
I think we should do the same as node https://github.com/projectkudu/KuduScript/blob/master/lib/templates/deploy.batch.node.template#L9

which is ignore this action if not defined.

**Comment by shrimpy** [comment]
ah... so this build script is also allow people to use outside of Azure. 
thanks for catching this! i didn`t realize it. will update soon.

**Pull request merged**